### PR TITLE
[BACKLOG-39678] - Schedule Output Perspective : Improve reporting engine to generate in line style and images for html rendering

### PR DESCRIPTION
**Pull request by soagarwal1**
@pentaho/hoth Please review

Part of PR set 
https://github.com/pentaho/pentaho-scheduler-plugin/pull/179
https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/244
https://github.com/pentaho/pentaho-reporting/pull/1657

### CODE REVIEW

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
<div align='center'>

[👍 Frogbot scanned this pull request and found that it did not add vulnerable dependencies.](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

<details>
<summary> <b>Note:</b> </summary>


---
<div align='center'>

**Frogbot** also supports **Contextual Analysis, Secret Detection, IaC and SAST Vulnerabilities Scanning**. This features are included as part of the [JFrog Advanced Security](https://jfrog.com/advanced-security) package, which isn't enabled on your system.

</div>


</details>


---
<div align='center'>

[🐸 JFrog Frogbot](https://docs.jfrog-applications.jfrog.io/jfrog-applications/frogbot)

</div>

**Comment by buildguy** [comment]
### :white_check_mark: Build finished in 7m 19s
Build command:
```console
mvn clean verify -B -e -Daudit -Djs.no.sandbox -pl core
```
---
:ok_hand: All tests passed!
 
Tests run: **540**, Failures: **0**, Skipped: **3** &nbsp;&nbsp; [Test Results](https://wingman.orl.eng.hitachivantara.com/job/projects/job/platform-reporting-plugin/job/PR-852/2/testReport)

 ---
 
 

:information_source:  _This is an automatic message_

File: core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java [hunk 1]
```diff
@@ -112,6 +112,9 @@ public class SimpleReportingAction implements IStreamProcessingAction, IStreamin
 
   public static final String RESERVEDMAPKEY_LINEAGE_ID = "lineage-id";
 
+  public static final String REPORT_EXTERNALIZE_STYLE = "externalize-style";
+  public static final String REPORT_BASE64_IMAGES = "base64-images";
```
Comment (singletonc) [inline core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java:116]: See comment related to these String constants https://github.com/pentaho/pentaho-scheduler-plugin-ee/pull/244/files#r1626298970

**Review by singletonc — APPROVED** [review]

File: core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java [hunk 2]
```diff
@@ -112,6 +112,9 @@ public class SimpleReportingAction implements IStreamProcessingAction, IStreamin
 
   public static final String RESERVEDMAPKEY_LINEAGE_ID = "lineage-id";
 
+  public static final String REPORT_EXTERNALIZE_STYLE = "externalize-style";
```
Comment (NJtwentyone) [inline core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java:115]: I'm fine with the variables. I'm worried about introducing a cyclic build.

The reporting projects - https://github.com/pentaho/pentaho-platform-plugin-reporting and https://github.com/pentaho/pentaho-reporting don't have an API jar.

I just discovered `"useJcr"` is a static constant - https://github.com/pentaho/pentaho-platform-plugin-reporting/blob/9b9b4a15605ce279d3321661f0dc061d3925832f/core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java#L102

File: core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java (hunk 1 above)
Comment (NJtwentyone) [reply core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java:116]: I'm fine with the variables. I'm worried about introducing a cyclic build between pentaho-scheduler and the reporting projects.

The reporting projects - https://github.com/pentaho/pentaho-platform-plugin-reporting and https://github.com/pentaho/pentaho-reporting don't have an API jar.

I just discovered "useJcr" is a static constant -

[pentaho-platform-plugin-reporting/core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java](https://github.com/pentaho/pentaho-platform-plugin-reporting/blob/9b9b4a15605ce279d3321661f0dc061d3925832f/core/src/main/java/org/pentaho/reporting/platform/plugin/SimpleReportingAction.java#L102)

Line 102 in [9b9b4a1](https://github.com/pentaho/pentaho-platform-plugin-reporting/commit/9b9b4a15605ce279d3321661f0dc061d3925832f)

 public static final String USE_JCR = "useJcr";

**Review by NJtwentyone — APPROVED** [review]

**Pull request merged**